webapp/channels/src/components/channel_layout/channel_controller.tsx

```diff
@@ -2,26 +2,28 @@
 // See LICENSE.txt for license information.
 
 import classNames from 'classnames';
-import React, {useEffect} from 'react';
+import React, {lazy, useEffect} from 'react';
 import {useDispatch, useSelector} from 'react-redux';
 
 import {cleanUpStatusAndProfileFetchingPoll} from 'mattermost-redux/actions/status_profile_polling';
 import {getIsUserStatusesConfigEnabled} from 'mattermost-redux/selectors/entities/common';
 
 import {addVisibleUsersInCurrentChannelToStatusPoll} from 'actions/status_actions';
 
+import {makeAsyncComponent} from 'components/async_load';
 import CenterChannel from 'components/channel_layout/center_channel';
 import LoadingScreen from 'components/loading_screen';
-import ProductNoticesModal from 'components/product_notices_modal';
-import ResetStatusModal from 'components/reset_status_modal';
-import Sidebar from 'components/sidebar';
 import CRTPostsChannelResetWatcher from 'components/threading/channel_threads/posts_channel_reset_watcher';
 import UnreadsStatusHandler from 'components/unreads_status_handler';
 
 import Pluggable from 'plugins/pluggable';
 import {Constants} from 'utils/constants';
 import {isInternetExplorer, isEdge} from 'utils/user_agent';
 
+const Sidebar = makeAsyncComponent('Sidebar', lazy(() => import('components/sidebar')));
```

Comment:
Same comment here as with the `ChannelIdentifierRouter` above.

I also wonder what the performance impact of lazily loading core components like these are. They're almost always mounted, so could the extra roundtrips to load them make the initial load slower?

Reply:
Separating them also separates the node_modules these require. But you are right we dont have to make this lazy